src/peer_conn_protocol.rs

```diff
@@ -5,32 +5,75 @@ use super::server;
 use std::io;
 use std::sync::mpsc::Sender;
 use uuid::Uuid;
+use std::collections::HashMap;
+use std::net::TcpStream;
 
-pub fn send_p2p_handshake(
-    conn_id: u64,
-    peer_address: String,
-    peer_port: u16,
-    messenger: Sender<MessengerOperations>,
-) -> Result<Uuid, io::Error> {
-    let stream = server::new_connection(peer_address.clone(), peer_port)?;
-    messenger
-        .send(MessengerOperations::New(NewConnectionMessage {
+pub struct Peer {
+    pub peer_address: String,
+    pub peer_port: u16,
+    pub stream: TcpStream,
+}
+
+impl Peer {
+    pub fn send_p2p_handshake(
+        &mut self,
+        conn_id: u64,
+        messenger: Sender<MessengerOperations>,
+    ) -> Result<Uuid, io::Error> {
+        
+        messenger
+            .send(MessengerOperations::New(NewConnectionMessage {
+                conn_id,
+                socket: self.stream.try_clone().unwrap(),
+            }))
+            .unwrap();
+    
+        send_packet!(
+            messenger,
             conn_id,
-            socket: stream.try_clone().unwrap(),
-        }))
+            Packet::Handshake(Handshake {
+                protocol_version: 404,
+                server_address: self.peer_address.clone(),
+                server_port: self.peer_port,
+                next_state: 6,
+            })
+        )
         .unwrap();
+    
+        Ok(Uuid::new_v4())
```

Comment:
Why does this and the function below return a uuid? Where is it used

Reply:
The uuid was left from the previous implementation I'll probably remove it. I thought we could get the server to call for this method (or maybe the handshake init protocol) because it has a unique connection id incremented for each message received. That's what I meant when I said we'll probably need to shift some code from the main.

Reply:
I don't think the server actually has anything--that connection ID variable is just on the stack of the server run method. We can't access it by calling the server module with the way it's currently implented I don't think

Reply:
Since the connection ID is passed when needed it should be accessible at anytime.

Reply:
So we're back to passing around the connection id into every function? If we're doing that then yeah there are no problems with uniqueness